src/supplicant/nm-supplicant-config.c

```diff
@@ -1057,6 +1057,9 @@ nm_supplicant_config_add_setting_8021x (NMSupplicantConfig *self,
 			fast = TRUE;
 			priv->fast_required = TRUE;
 		}
+
+		if (method && num_eap == 1 && strcasecmp (method, "external") == 0)
```

Comment:
using `strcasecmp()` here is most certainly wrong, because `NMSetting8021x::verify()` ensures that the method values matches literally (case-sensitive). Existing code should be fixed, but at least new code should not get it wrong too.

In fact, using `strcasecmp()` in NetworkManager daemon is always wrong, because the function is locale dependent. If case-insensitive comparison is needed, `g_ascii_strcasecmp()` would be better. But that is not the case here.

Teh code here rejects "external" only, if it is present alone in the list of EAP methods (good). But the lines above with `ADD_STRING_LIST_VAL()` should filter out "external". As we anyway iterate over `num_eap` here, it seems simpler to drop `ADD_STRING_LIST_VAL()` macro and just create the supplicant option here (thereby skipping over "external").

Also, you cannot `return FALSE`, without also setting `@error`.